src/business/products/add-products.ts

```diff
@@ -0,0 +1,135 @@
+  getUserByAttribute,
+  insertProduct,
+} from "../../repository/firestore/index.js";
+import { postProducts } from "../../repository/woo-api/create-post-product.js";
+
+import type {
+  Request,
+  Response,
+} from "express";
+
+dotenv.config();
+
+const SERVICE_ERRORS = {
+  notAuthorized: {
+    statusCode: StatusCodes.UNAUTHORIZED,
+    type: "/products/not-authorized",
+    message: "not authorized",
+  },
+  resourceNotFound: {
+    statusCode: StatusCodes.NOT_FOUND,
+    type: "/products/not-found",
+    message: "resource not found",
+  },
+  notAllowed: {
+    statusCode: StatusCodes.FORBIDDEN,
+    type: "/products/not-allowed",
+    message: "query missing or not allowed",
+  },
+  invalidRequestType: {
+    statusCode: StatusCodes.BAD_REQUEST,
+    type: "/products/add-product-failed",
+    message: "request body is not valid",
+  },
+  existingProduct: {
+    statusCode: StatusCodes.BAD_REQUEST,
+    type: "/products/add-product-failed",
+    message: "product name exist",
+  },
+};
+
+const AddProductRequest = Type.Object({
+  name: Type.String(),
+  sku: Type.String(),
+  price: Type.String(),
+  stock_quantity: Type.Union([ Type.Number() ], Type.Null()), // string || null
+  // Type.Optional(Type.Union([Type.Null(), Type.String()])) // string || null || undefined
+  images: Type.Array(
+    Type.Object({
+      id: Type.Number(),
+      src: Type.String(),
+    }),
+  ),
+});
+
+export const addProducts = async (req: Request, res: Response) => {
+  const productDetails = req.body;
+  const isAddProductTypeRequestValid = isResponseTypeTrue(
+    AddProductRequest,
+    req.body,
+    false,
+  );
+
+  if (!isAddProductTypeRequestValid.isValid) {
+    logger.log(
+      "error",
+      `invalid add product request type  ${
+        isAddProductTypeRequestValid.errors[0]?.message
+      } **Expected** ${JSON.stringify(
+        AddProductRequest,
+      )} **RECEIVED** ${JSON.stringify(req.body)}`,
+    );
+    return createErrorResponse(res, SERVICE_ERRORS.invalidRequestType);
+  }
+
+  // TODO return error if product name already exist
+
+  const userId = createVerifyBasicAuthHeaderToken(req.headers["authorization"]);
+
+  if (!userId) {
+    logger.log(
+      "error",
+      `no decoded token from ${JSON.stringify(
+        req.headers["authorization"],
+      )} authorization header`,
+    );
+    return createErrorResponse(res, SERVICE_ERRORS.notAllowed);
+  }
+
+  const userFoundInFirestore = await getUserByAttribute("user_id", userId);
+  if (!userFoundInFirestore) {
+    logger.log("error", `user not found by id ${userId}`);
+    return createErrorResponse(res, SERVICE_ERRORS.resourceNotFound);
+  }
+
+  const wooBasicAuth = createBasicAuthHeaderToken(
+    userFoundInFirestore.woo_credentials.token,
+    userFoundInFirestore.woo_credentials.secret,
+  );
+
+  const base_url =
+    process.env["NOVE_ENV"] === "product"
+      ? userFoundInFirestore.store.app_url
+      : (process.env["WOO_BASE_URL"] as string);
+
+  try {
```

Comment:
why try catch? we wrap the function in try - catch already. look 
[https://github.com/Woo-Pick-Fullfillment-Inventory/Backend-Woo-Inventory/pull/53/files#diff-64fd237c06fdb42ecf1266b0e420cd4216918363f27b8af8afa425361adb2578R12)]